nailed-it/New Group/PolishLibraryViewController.swift

```diff
@@ -189,59 +238,117 @@ extension UIImage {
 
 extension PolishLibraryViewController: CZPickerViewDelegate, CZPickerViewDataSource {
     func czpickerViewWillDisplay(_ pickerView: CZPickerView!) {
-        pickerView.setSelectedRows(self.selectedRows)
+        if self.sortFilter == "filter" {
+            pickerView.setSelectedRows(self.selectedRows)
+        }
     }
 
     func numberOfRows(in pickerView: CZPickerView!) -> Int {
-        return self.brands.count
+        if self.sortFilter == "filter" {
+            return self.brands.count
+        } else {
+            return self.sortingOptions.count
+        }
     }
 
     func numberOfRowsInPickerView(pickerView: CZPickerView!) -> Int {
-        return self.brands.count
+        if self.sortFilter == "filter" {
+            return self.brands.count
+        } else {
+            return self.sortingOptions.count
+        }
     }
 
     func czpickerView(_ pickerView: CZPickerView!, titleForRow row: Int) -> String! {
-        return self.brands[row]
-    }
+        if self.sortFilter == "filter" {
+            return self.brands[row]
+        } else {
+            return self.sortingOptions[row]
+        }
 
-    func czpickerView(_ pickerView: CZPickerView!, didConfirmWithItemAtRow row: Int){
-        self.navigationController?.setNavigationBarHidden(true, animated: true)
     }
 
     func czpickerViewDidClickCancelButton(_ pickerView: CZPickerView!) {
         self.navigationController?.setNavigationBarHidden(false, animated: true)
     }
+    
+    func czpickerView(_ pickerView: CZPickerView!, didConfirmWithItemAtRow row: Int){
+        if self.sortFilter == "sort" {
+            if self.sortingOptions[row] == "Price: $ to $$$" {
+                startAnimating(size, message: "Sorting...", type: NVActivityIndicatorType.ballTrianglePath)
+                // Sorting based on brand right now, because it corresponds to price
+                let sortedColors = self.colors?.sorted {
+                    let string0 = String(describing: $0.brand)
+                    let string1 = String(describing: $1.brand)
+                    return string0 > string1
+                }
+                self.stopAnimating()
```

Comment:
these next 3 lines are repeated in almost every block, can we consolidate into a funciton somehow?